functional_offset_manager_test.go

```diff
@@ -24,10 +19,6 @@ func TestFuncOffsetManager(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if _, err := offsetManager.ManagePartition("does_not_exist", 123); err != ErrUnknownTopicOrPartition {
-		t.Fatal("Expected ErrUnknownTopicOrPartition when starting a partition offset manager for a partition that does not exist, got:", err)
-	}
-
```

Comment:
You no longer get back `ErrUnknownTopicOrPartition`, you get back `ErrNoError` with an offset of `-1` (just like you do for topic/partitions that exist but haven't yet had an offset committed). I'm honestly not sure if this is a kafka bug, or an intended behaviour change.